ext/vulnsrc/rhel/rhel.go

```diff
@@ -184,18 +185,31 @@ func parseRHSA(ovalReader io.Reader) (vulnerabilities []database.VulnerabilityWi
 	for _, definition := range ov.Definitions {
 		pkgs := toFeatures(definition.Criteria)
 		if len(pkgs) > 0 {
+
+			// Init vulnerability
 			vulnerability := database.VulnerabilityWithAffected{
 				Vulnerability: database.Vulnerability{
-					Name:        name(definition),
-					Link:        link(definition),
+					Name:        rhsaName(definition),
+					Link:        rhsaLink(definition),
 					Severity:    severity(definition),
 					Description: description(definition),
 				},
 			}
 			for _, p := range pkgs {
 				vulnerability.Affected = append(vulnerability.Affected, p)
 			}
-			vulnerabilities = append(vulnerabilities, vulnerability)
+
+			// Only RHSA is present
+			if len(definition.References) == 1 {
+				vulnerability.Link = definition.References[0].URI
+				vulnerabilities = append(vulnerabilities, vulnerability)
+			} else {
+				for _, reference := range definition.References[1:] {
```

Comment:
We don't have to test if (ref.Source == "CVE") since all the references starting from the index 1 are CVE. 
We create one vulnerability for each CVE. 
As append copies the vulnerability, we can reuse the same object for each loop and only change the name and the link.